srunner/scenarios/parking_exit.py

```diff
@@ -0,0 +1,174 @@
+                                          world,
+                                          debug_mode,
+                                          criteria_enable=criteria_enable)
+
+    def _create_behavior(self):
+        """
+        Deactivate OutsideRouteLanesTest, then move ego to the parking point, generate blocking vehicles in front of and behind the ego.
+        After ego drives away, actviate OutsideRouteLanesTest, end scenario.
+        """
+
+        sequence = py_trees.composites.Sequence()
+
+        parking_transform = front_transform = carla.Transform()
+        parking_transform.location = self._parking_point
+        # To get rotation like this because the return value of the
+        # get_waypoint(self, location, project_to_road=True, lane_type=carla.LaneType.Parking)
+        # would be weird in some maps. So using get_waypoint(self._parking_point) to get a waypoint in driving lane
+        # which has the same rotation with the parking point is a safe choice.
+        parking_transform.rotation = self._map.get_waypoint(
+            self._parking_point).transform.rotation
+
+        # Put blocking vehicles
+        front_transform = get_same_dir_transform_by_distance(
+            parking_transform, self._front_vehicle_distance)
+        front_transform.location.z += 0.5
+        behind_transform = get_same_dir_transform_by_distance(
+            parking_transform, -1*self._behind_vehicle_distance)
+        behind_transform.location.z += 0.5
+
+        actor_source_front = ActorSource(
+            ['vehicle.audi.tt', 'vehicle.tesla.model3', 'vehicle.nissan.micra'],
+            front_transform, 1, "ParkingExit_front_transform_queue", 1)
+
+        actor_source_behind = ActorSource(
+            ['vehicle.audi.tt', 'vehicle.tesla.model3', 'vehicle.nissan.micra'],
+            behind_transform, 1, "ParkingExit_front_transform_queue", 1)
+
+        # Deactivate OutsideRouteLanesTest
+        sequence.add_child(SwitchOutsideRouteLanesTest(False))
+
+        # Teleport ego to the parking point
+        sequence.add_child(ActorTransformSetter(
+            self.ego_vehicles[0], parking_transform))
+
+        root = py_trees.composites.Parallel(
+            policy=py_trees.common.ParallelPolicy.SUCCESS_ON_ONE)
+        root.add_child(actor_source_front)
+        root.add_child(actor_source_behind)
+
+        end_condition = py_trees.composites.Sequence()
+        end_condition.add_child(InTriggerDistanceToLocation(
+            self.ego_vehicles[0], self._parking_point, self._end_distance, operator.gt, name="EndTrigger"))
+        root.add_child(end_condition)
+        sequence.add_child(root)
+
+        sequence.add_child(SwitchOutsideRouteLanesTest(True))
+
+        return sequence
+
+    def _create_test_criteria(self):
+        """
+        A list of all test criteria will be created that is later used
+        in parallel behavior tree.
+        """
+        return [CollisionTest(self.ego_vehicles[0])]
```

Comment:
I forgot to mention this in the previous scenario. For routes, the criteria should be empty, as the collisiontest is already part of the whole route. Add this 

```
if self.route_mode:
    return[]
```

Reply:
Added